compiler/x/codegen/ControlFlowEvaluator.cpp

```diff
@@ -491,22 +491,23 @@ TR::Register *OMR::X86::TreeEvaluator::tableEvaluator(TR::Node *node, TR::CodeGe
       }
 
    TR::X86MemTableInstruction *jmpTableInstruction = NULL;
+   TR::RegisterDependencyConditions *deps = NULL;
 
-   if (cg->getLinkage()->getProperties().getMethodMetaDataRegister() != TR::RealRegister::NoReg)
```

Comment:
@0xdaryl was this the condition that was somehow only allowing this logic for Java ?

Reply:
Yes.  I don't like it and it will be changed in the future.

Reply:
Just to be clear, the meta data register is only set to a non-NoReg register on OpenJ9-specific linkages.